src/NHibernate.Test/Tools/hbm2ddl/SchemaMetadataUpdaterTest/SchemaMetadataUpdaterFixture.cs

```diff
@@ -56,10 +57,94 @@ public void UpdateReservedWordsInDialect()
 			var sf = (ISessionFactoryImplementor) configuration.BuildSessionFactory();
 			SchemaMetadataUpdater.Update(sf);
 			var match = reservedDb.Intersect(sf.Dialect.Keywords);
+
+			// tests that nothing in the first metaData.GetReservedWords() is left out of the second metaData.GetReservedWords() call.
+			// i.e. always passes.
+			Assert.That(match, Is.EquivalentTo(reservedDb));
+		}
+
+		[Test]
+		public void EnsureReservedWordsHardCodedInDialect()
+		{
+			var reservedDb = new HashSet<string>();
+			var configuration = TestConfigurationHelper.GetDefaultConfiguration();
+			var dialect = Dialect.Dialect.GetDialect(configuration.Properties);
+			var connectionHelper = new ManagedProviderConnectionHelper(configuration.Properties);
+			connectionHelper.Prepare();
+			try
+			{
+				var metaData = dialect.GetDataBaseSchema(connectionHelper.Connection);
+				foreach (var rw in metaData.GetReservedWords())
+				{
+					var rwl = rw.ToLowerInvariant();
+					if (rwl.Contains(", version=")) continue;
+					reservedDb.Add(rwl);
+				}
+			}
+			finally
+			{
+				connectionHelper.Release();
+			}
+
+			var sf = (ISessionFactoryImplementor)configuration.BuildSessionFactory();
+
+			// use the dialect as configured, with no update
+			var match = reservedDb.Intersect(sf.Dialect.Keywords).ToList();
+
+			// tests that nothing in metaData.GetReservedWords() is left out of the Dialect.Keywords (without a refresh).
+			var differences = reservedDb.Except(match).ToList();
+			if (differences.Count > 0)
+			{
+				Console.WriteLine("Update Dialect {0} with RegisterKeyword:", sf.Dialect.GetType().Name);
+				foreach (var keyword in differences.OrderBy(x => x))
+				{
+					Console.WriteLine("  RegisterKeyword(\"{0}\");", keyword);
+				}
+			}
+
 			Assert.That(match, Is.EquivalentTo(reservedDb));
 		}
 
 		[Test]
+		public void CheckForExcessReservedWordsHardCodedInDialect()
```

Comment:
Is this test relevant? It looks like the approach taken to solve this issue is to set keywords on old dialect like SQL2000, while using indeed a more recent server likely having more keywords, like SQL2016 (what we have on the build agent). So indeed testing this against an actual SQL Server 2000 would likely yield some excess words. Does it matter much? I do not think. So I do not think this test is useful.

(Do not try to emulate this with SQL Server compatibility level: that is just a backward compatibility setting, re-enabling obsolete features but not disabling newer ones. For actually testing a SQL Server 2000, only an actual SQL Server 2000 is suitable.)